src/parser/demonhunter/havoc/modules/talents/Demonic.js

```diff
@@ -0,0 +1,106 @@
+import React from 'react';
+import SPELLS from 'common/SPELLS';
+import TalentStatisticBox from 'interface/others/TalentStatisticBox';
+import STATISTIC_ORDER from 'interface/others/STATISTIC_ORDER';
+import Events from 'parser/core/Events';
+import Analyzer, { SELECTED_PLAYER } from 'parser/core/Analyzer';
+import SpellLink from 'common/SpellLink';
+
+/**
+ * Example Report: https://www.warcraftlogs.com/reports/23dHWCrT18qhaJbz/#fight=1&source=16
+ */
+
+const META_BUFF_DURATION_EYEBEAM = 10000;
+
+class Demonic extends Analyzer{
+
+  eyeBeamCasts = 0;
+  goodDeathSweep = 0;
+  eyeBeamTimeStamp = undefined;
+  deathsweepsInMetaCounter = undefined;
+  badCasts = 0;
+
+  constructor(...args) {
+    super(...args);
+    this.active = this.selectedCombatant.hasTalent(SPELLS.DEMONIC_TALENT.id);
+    if (!this.active) {
+      return;
+    }
+    this.addEventListener(Events.cast.by(SELECTED_PLAYER).spell(SPELLS.EYE_BEAM), this.onEyeBeamCast);
+    this.addEventListener(Events.cast.by(SELECTED_PLAYER).spell(SPELLS.DEATH_SWEEP), this.onDeathSweepCast);
+  }
+
+  onEyeBeamCast(event) {
+    const hasMetaBuff = this.selectedCombatant.hasBuff(SPELLS.METAMORPHOSIS_HAVOC_BUFF.id, event.timestamp - 1000);
+
+    if (hasMetaBuff) {
+      return;
+    }
+    this.eyeBeamCasts += 1;
+    this.eyeBeamTimeStamp = event.timestamp;
+
+    if (this.deathsweepsInMetaCounter === undefined) {
+      this.deathsweepsInMetaCounter = 0;
+      return;
+    }
+
+    if (this.deathsweepsInMetaCounter < 2) {
+      this.badCasts += 1;
+    }
+
+    this.deathsweepsInMetaCounter = 0;
+  }
+
+  onDeathSweepCast(event) {
+    if (this.eyeBeamTimeStamp !== undefined && (event.timestamp - this.eyeBeamTimeStamp) < META_BUFF_DURATION_EYEBEAM) {
+      this.goodDeathSweep += 1;
+      this.deathsweepsInMetaCounter += 1;
+    }
+  }
+
+  get avgDeathSweepPerEyeBeam(){
+    //Devide by 0 check incase they never cast Death Sweep
```

Comment:
You can do `return (this.goodDeathSweep / this.eyeBeamCasts) || 0;`? If goodDeathSweep is 0, then regardless of eyeBeamCasts it returns 0. If eyeBeamCasts is 0, then this whole thing evaluates to I think `Infinity` and `|| 0` makes it return 0 too. Otherwise it's the average DS per Beam.

Reply:
done thanks!